### ISSUE TITLE
Auto finish/Auto-start daily glitch

### ISSUE THREAD

**Issue by kaeroku** [genbtc/AutoTrimps#87]
Issue Summary: AT appears to be substituting 100% He bonus for actual dailies, and not rewarding correct/expected He at the end of runs, even with Auto-finish Daily enabled.

 'Current challenge' displays +100% only and doesn't have expected daily effects applied to run (such as enemy attack increases, etc.) Also expected Helium bounty from challenges (300%+) is not seen; rather, ~100% is seen. 

There's two sources of information I have for helping track this down. First, images sourced from my own run demonstrating the error. The second is a link to a reddit post comment by GreenSatellite I found while I was trying to track down the issue. *The post is about a different (but I believe related) issue.*

Link to the post: https://www.reddit.com/r/Trimps/comments/77ekwd/dailies_awarding_too_much_helium/doo6kr4/

![perks screen](https://user-images.githubusercontent.com/18428099/32031672-ae93ef64-b9d0-11e7-8f2c-24043f1b56d0.JPG)
![most likely relevant at settings](https://user-images.githubusercontent.com/18428099/32031690-cc6734ce-b9d0-11e7-9730-6a832f934ac9.JPG)
![125 total before daily vms finished](https://user-images.githubusercontent.com/18428099/32031779-37f67722-b9d1-11e7-9f73-c8d1ac1bb2ba.JPG)
![126 total after daily finished portal](https://user-images.githubusercontent.com/18428099/32031780-380275ae-b9d1-11e7-9ced-d1b7a2fbd710.JPG)
![daily info for the run indicated](https://user-images.githubusercontent.com/18428099/32031781-381369b8-b9d1-11e7-9f7b-a6e2fcd5e874.JPG)

For the last two images, I'm going to attempt to highlight why I think other daily effects are not being applied correctly.

Expected: According to the current daily that is supposedly being run by AT auto-selecting the daily challenge, there should be loot stacks accumulating *somewhere,* one would assume on the combat screen where most stacks are indicated, to show an increase in non-He loot. (See images below.) 
Actual result: no stacks show on combat screen, nothing shows in the breakdown for current loot, there's absolutely no indicator that the current daily is being run. However, at the end of my run the daily is no longer available.

![current run combat screen after insta-killing a couple dozen imps](https://user-images.githubusercontent.com/18428099/32031922-3612bc3a-b9d2-11e7-8232-aa75c63976c3.JPG)
![current run daily info](https://user-images.githubusercontent.com/18428099/32031923-36239096-b9d2-11e7-94a6-7268d61e18db.JPG)
![current run loot](https://user-images.githubusercontent.com/18428099/32031924-3634be02-b9d2-11e7-8322-b9fd8cbd072e.JPG)

I'm not experienced enough to dig through and figure out why this is happening, but something seems to be wrong with the way AT is handling auto-dailies.

**Comment by kaeroku** [comment]
New info: Note that if I manually click "finish daily" during a run, it appears to self-correct for the *next* run, but then returns to the failure mode when completing a run by itself. Also note that if I manually click "finish daily" during a run, it appears to not correctly follow the "auto-allocate perks" setting, but will return to correctly auto-allocating perks *when it is in failure mode (for auto-portaling) only.*

**Comment by genbtc** [comment]
Please someone get on top of understanding this issue - I dont have the time right now but this seems important

**Comment by genbtc** [comment]
I have just revisited your description of this issue, and I still conclude that it is very weird. Heres my daily. And all my dailies have always launched with the proper mods and everything. Theres not really much "to it" its handled by game code usually.  Sounds like its possible that you are going into a daily that doesnt really exist.
![chrome_2017-11-11_21-05-05](https://user-images.githubusercontent.com/3683744/32695254-3eed874a-c724-11e7-8f80-14631832ee8b.png)

Note that Auto Allocate Perks only allocates on autoportal.

**Comment by kaeroku** [comment]
My understanding re: Auto-allocate perks matches yours. Issue was very strange.

I've run several dailies since posting this to see if the bug was repeatable, and it has been. All I have to do is set dailies to auto-run.

Note that the bug doesn't occur if I manually enter dailies, it only occurs after it auto-portals.

I'll try to recreate the issue, since I know you've done some edits to the code since I posted this, and see if it's still happening. Give me... oh, a day or two.

**Comment by kaeroku** [comment]
Okay. This should be all relevant settings. Ending a Lead run on this tab now:

![image](https://user-images.githubusercontent.com/18428099/32695788-2c7db804-c734-11e7-8bd9-90229e66e057.png)

Here are the available challenges on this tab: 

![image](https://user-images.githubusercontent.com/18428099/32695792-569eb854-c734-11e7-9c1d-2a5441ca503c.png)

As you can see, no dailies have been run. I'm pretty sure this daily will be a pain for me to run without major settings changes, so I'm going to modify one setting before starting: allowing it to auto-portal super early, so the run can be tested without major time investment. In order to do this, I'm changing "Finish Daily Zone Mod" to -100, which ought to allow AT to portal on or after z81, given the other settings. 

Lead currently takes about 3.5 hours on this tab and has been running for 2h so far, so the daily should start under the current settings sometime in the next 1.5-2 hrs. I would enter the Daily manually now, except that manually starting Dailies hasn't caused the failure mode to occur. 

I'm pretty sure that the failure mode doesn't actually occur when the Daily auto-starts, I'm *pretty* sure that it occurs when the daily auto-finishes. But I'll double check that with this run. (If auto-finish is the trigger, we won't see failure until it starts the 2nd daily, which I won't be awake to see and won't have info on until tomorrow. If auto-start is the trigger, I should have data for you in about 2 hrs.)

**Comment by kaeroku** [comment]
Just occurred to me; I had assumed that auto-start/auto-finish Dailies would revert to whatever He challenge you had set if there were no dailies available. This may have been a mistaken assumption. Do I need to have He/hr selected and "none" for challenges in order for dailies to complete correctly? If so, it's not a bug as much as a documentation... misunderstanding.

**Comment by kaeroku** [comment]
Also helpful would be a screenshot of your settings when you get successful dailies going, would help me narrow down what the cause of the conflict may be and why I get it and you don't.

**Comment by kaeroku** [comment]
First Daily after Lead behaves as expected (that is, correctly): 

![image](https://user-images.githubusercontent.com/18428099/32696374-361c2d2c-c744-11e7-9c06-c03af748d8ed.png)

Note: _this is expected behavior._ In OP I suspected the issue was on auto-finish. I'll have to give this run some time to complete. I'm going to watch it until z100 or so, but given it's set to end when He/hr drops, it could be a couple hours and I need to sleep soon.

**Comment by kaeroku** [comment]
Hmmm. Second daily completed correctly and started another daily correctly.

I don't know what bug I ran into before, but it appears that it either was incidentally resolved, or I've reached the point where I'm no longer encountering failure mode.

This might be able to be closed. Though, I'm half way tempted to find someone with an active Nom save, put that in a new tab and try running it (as that's when I encountered this originally.)

Edit: Trimps received an update on 11/3 as well, which was after this issue was opened. Quite possible the issue was resolved then, as I haven't run dailies since then, although it is not showing in the bug fixes.
